### PR TITLE
fix: fix styling issues with algolia search

### PR DESCRIPTION
**Pull request by Giovanni-Schroevers**

### CODE REVIEW

File: examples/magento-graphcms/pages/search/index.tsx [hunk 1]
```diff
@@ -93,11 +95,21 @@ function SearchResultPage(props: SearchResultProps) {
...
+            ))}
+          </CategorySearchResults>
         </Container>
 
+        <SearchDivider
+          sx={(theme) => ({
+            [theme.breakpoints.down('md')]: {
+              display: 'none',
+            },
+          })}
```
Comment (paales) [inline examples/magento-graphcms/pages/search/index.tsx:110]:
```suggestion
          sx={{ display: { xs: 'none', md: 'block' } }}
```

File: packages/algolia-search/plugins/AlgoliaCategorySearchResults.tsx [hunk 2]
```diff
@@ -26,19 +27,19 @@ function CategoryHits() {
   )
 }
 
-function AlgoliaCategorySearchPlugin(props: PluginProps<SearchFormProps>) {
+function AlgoliaCategorySearchPlugin(props: PluginProps<CategorySearchResultsProps>) {
   const { Prev, ...rest } = props
```
Comment (paales) [inline packages/algolia-search/plugins/AlgoliaCategorySearchResults.tsx:31]:
```suggestion
function AlgoliaCategorySearchPlugin(props: PluginProps<CategorySearchResultsProps>) {
  const { Prev, children, ...rest } = props
```

File: packages/algolia-search/plugins/AlgoliaCategorySearchResults.tsx [hunk 3]
```diff
@@ -26,19 +27,19 @@ function CategoryHits() {
...
   const searchIndex = useAlgoliaSearchIndexConfig('_categories')?.searchIndex
 
   if (!searchIndex) return <Prev {...rest} />
 
   return (
-    <>
-      <Prev {...rest} />
+    <Prev {...rest}>
+      {rest.children}
```
Comment (paales) [inline packages/algolia-search/plugins/AlgoliaCategorySearchResults.tsx:38]:
```suggestion
      {children}
```

File: packages/algolia-search/plugins/AlgoliaPageSearchResults.tsx [hunk 4]
```diff
@@ -34,19 +34,19 @@ function PageHits() {
   )
 }
 
-function AlgoliaPageSearchPlugin(props: PluginProps<SearchFormProps>) {
+function AlgoliaPageSearchPlugin(props: PluginProps<CategorySearchResultsProps>) {
```
Comment (paales) [inline packages/algolia-search/plugins/AlgoliaPageSearchResults.tsx:37]: Should we also rename the plugin to AlgoliaCategorySearchResultsPlugin?

**Review by paales — CHANGES_REQUESTED** [review]

File: packages/algolia-search/plugins/AlgoliaPageSearchResults.tsx (hunk 4 above)
Comment (Giovanni-Schroevers) [reply packages/algolia-search/plugins/AlgoliaPageSearchResults.tsx:37]: This plugin is still responsible for the page index so I think the current name should be more applicable

**Comment by paales** [comment]
@Giovanni-Schroevers I'm fine with merging this. Do you want to wait for the review of @mikekeehnen?

**Comment by Giovanni-Schroevers** [comment]
@paales Seems fine to me as well but we aren't planning on doing another upgrade for a while so it is in no rush so if you would like to have @mikekeehnen take a look first we can wait on that

File: packages/algolia-search/components/SearchBox/SearchBox.tsx [hunk 5]
```diff
@@ -51,7 +51,7 @@ export function SearchBox(props: SearchBoxProps) {
       inputRef={searchInputElement}
       onChange={debounceSearch}
       fullWidth
-      sx={{ mt: 1, mb: 1 }}
+      sx={[{ mt: 1, mb: 1 }, ...(Array.isArray(sx) ? sx : [sx])]}
```
Comment (mikekeehnen) [inline packages/algolia-search/components/SearchBox/SearchBox.tsx:54]: Are the margins still relevant after using the `CategorySearchResults` component on the search page?

**Review by mikekeehnen — DISMISSED** [review]

File: packages/magento-search/components/CategorySearchResult/CategorySearchResults.tsx
Comment (mikekeehnen) [inline packages/magento-search/components/CategorySearchResult/CategorySearchResults.tsx:1]: This component looks like it doesn't really have a purpose besides using it as anchor point for the plugin. Is this really necessary?

File: packages/magento-search/components/CategorySearchResult/CategorySearchResult.tsx [hunk 6]
```diff
@@ -53,7 +52,6 @@ export function CategorySearchResult(props: CategorySearchResultProps) {
               text={breadcrumb?.category_name ?? ''}
               highlight={search}
             />
-            {' / '}
```
Comment (mikekeehnen) [inline packages/magento-search/components/CategorySearchResult/CategorySearchResult.tsx:56]: You may want to still have a `/` if the breadcrumbs show a path thats deeper then 1 level. Maybe only show a `/` if `length > 1` and `index > 0` ?

File: packages/magento-search/components/CategorySearchResult/CategorySearchResults.tsx
Comment (Giovanni-Schroevers) [reply packages/magento-search/components/CategorySearchResult/CategorySearchResults.tsx:1]: To me this seemed like a better solution than to abuse the `SearchForm` component for this. Using that component also gave some unintended styling changes to for example the search on QPlus

File: packages/algolia-search/components/SearchBox/SearchBox.tsx (hunk 5 above)
Comment (Giovanni-Schroevers) [reply packages/algolia-search/components/SearchBox/SearchBox.tsx:54]: We still want this margin if there are no search results so I would say yes

**Review by mikekeehnen — APPROVED** [review]

**Pull request merged**